docs/src/man/iterative_refinement.md

```diff
@@ -11,61 +12,213 @@ Depth = 3
+DIR_EAST, DIR_NORTH, DIR_WEST, DIR_SOUTH = [1., 0.], [0., 1.], [-1., 0.], [0., -1.]
+pe, pn, pw, ps = σ(DIR_EAST, b), σ(DIR_NORTH, b), σ(DIR_WEST, b), σ(DIR_SOUTH, b)
+
+addapproximation!(Ω, pn, DIR_NORTH, pw, DIR_WEST)
+addapproximation!(Ω, pw, DIR_WEST, ps, DIR_SOUTH)
+addapproximation!(Ω, ps, DIR_SOUTH, pe, DIR_EAST)
+
+plot!(Approximations.tohrep(Ω), alpha=0.2, color="orange")
+```
+
+Next we refine the first approximation of the list.
+
+```@example example_iterative_refinement
+import LazySets.Approximations.refine
+
+(r1, r2) = refine(Ω, 1)
+Ω.approx_list[1] = r1
+insert!(Ω.approx_list, 2, r2)
+
+plot(b, 1e-3, aspectratio=1, alpha=0.3)
+plot!(Approximations.tohrep(Ω), alpha=0.2, color="orange")
 ```
 
-Let's analyze in some detail how `p` was actually computed. The "box" directions
-are used as the initialization.
+We call `r1` and `r2` the right and left approximations respectively, since
+they are saved in counter-clockwise order. We can check that the first two
+approximations are still refinable.
+
+```@example example_iterative_refinement
+Ω.approx_list[1].refinable,  Ω.approx_list[2].refinable
+```
 
-```@example ir
-import LazySets.Approximations:Approximation2D, refine
+Hence, we can make again a refinement of that approximation.
 
-const DIR_EAST, DIR_NORTH, DIR_WEST, DIR_SOUTH = [1., 0.], [0., 1.], [-1., 0.], [0., -1.]
+```@example example_iterative_refinement
+(r1, r2) = refine(Ω, 1)
+Ω.approx_list[1] = r1
+insert!(Ω.approx_list, 2, r2)
 
+plot(b, 1e-3, aspectratio=1, alpha=0.3)
+plot!(Approximations.tohrep(Ω), alpha=0.2, color="orange")
 ```
+
+The criterion for an approximation being refinable is that we can properly define
+a normal direction `ndir`. This boils down to checking for the following
+"degenerate" cases:
+
+1. ``p_1`` and ``p_2`` overlap.
+2. ``p_1`` and ``q`` overlap.
+1. ``p_2`` and ``q`` overlap.
+
+Moreover, we include the condition `approx_error > TOL` where `TOL` is the floating
+point epsilon in the given numerical precision.
+
+## Algorithm
+
+Having presented the individual steps, we give the pseudocode of the iterative
+refinement algorithm, see `approximate(S, ε)`.
+
+The algorithm consists of the following steps:
+
+1. *Initialization*. The approximation is initialized with box directions,
+   i.e. it starts with four `LocalApproximation` objects. Let `i=1`.
+2. *Refinement loop*. If the local approximation at index `i` has an error greater
+   than the threshold `ε`, then refine. Otherwise, increment `i <- i+1`.
+3. *Redundancy check*. Insert the refined right approximation at position `i`,
+   and check whether the left approximation is redundant or not with respect to the one
+   at position `i+1`. Checking for redundancy amounts to checking for overlap of both
+   `p1` and `q`. Then, either substitute at `i+1` or insert (keeping the approximation
+    at `i+1`) depending on the redundancy check.
+4. *Stopping criteria*. If the index `i` is smaller or equal than the current length of
```

Comment:
"criterion"?
Currently it is rather a criterion when to *not* stop. What about:
```
*Stopping criterion*. Terminate if the index `i` exceeds the current length of the approximations list; otherwise continue with step 2.
```

Reply:
yes this is better, thx.